torch/ao/sparsity/experimental/data_sparsifier/lightning/tests/test_callbacks.py

```diff
@@ -78,5 +105,121 @@ def test_post_training_callback(self):
         self.run_all_checks(pl_module, callback, sparsifier_args)
 
 
+class _TrainingAwareCallbackTestCase(TestCase):
+    """Class to test in-training version of lightning callback
+    Simulates model training and makes sure that each hook is doing what is expected
+    """
+    def _check_on_train_start(self, pl_module, callback, sparsifier_args, scheduler_args):
+
+        callback.on_train_start(42, pl_module)  # 42 is a dummy value
+
+        # sparsifier and scheduler instantiated
+        assert callback.data_scheduler is not None and callback.data_sparsifier is not None
+
+        for key, value in sparsifier_args.items():
+            callback.data_sparsifier.defaults[key] == value
+
+        for key, value in scheduler_args.items():
+            assert getattr(callback.data_scheduler, key) == value
+
+    def _simulate_update_param_model(self, pl_module):
+        """This function might not be needed as the model is being copied
+        during train_epoch_end() but good to have if things change in the future
+        """
+        for _, param in pl_module.model.named_parameters():
+            param.data = param + 1
+
+    def _check_on_train_epoch_start(self, pl_module, callback):
+        # check if each component of state dict is being loaded correctly
+        callback.on_train_epoch_start(42, pl_module)
+        if callback.data_sparsifier_state_dict is None:
+            return
+
+        data_sparsifier_state_dict = callback.data_sparsifier.state_dict()
+
+        # compare container objects
```

Comment:
nit: can we give a bit more detail on these comparisons? e.g. why do we compare them, what is the expected result

Reply:
done